src/main/java/ai/shreds/infrastructure/InfrastructurePurchaseTransactionRepositoryImpl.java

```diff
@@ -0,0 +1,37 @@
+package ai.shreds.infrastructure;
+
+import ai.shreds.domain.DomainPurchaseTransactionEntity;
+import ai.shreds.domain.DomainPurchaseTransactionPort;
+import ai.shreds.infrastructure.mapper.InfrastructureEntityMapper;
+import ai.shreds.infrastructure.repository.PurchaseTransactionJpaRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class InfrastructurePurchaseTransactionRepositoryImpl implements DomainPurchaseTransactionPort {
+
+    private static final Logger logger = LoggerFactory.getLogger(InfrastructurePurchaseTransactionRepositoryImpl.class);
+
+    private final PurchaseTransactionJpaRepository purchaseTransactionJpaRepository;
```

Comment:
Add the PurchaseTransactionJpaRepository interface to the service with complete code